src/main/resources/com/google/api/codegen/java/smoke_test.snip

```diff
@@ -24,32 +24,31 @@
     @else
       {@executeWithNoProjectID(smokeTest)}
     @end
+    {@getProjectIdForTest()}
   }
 @end
 
 @private testBody(smokeTest)
   @if smokeTest.requireProjectId
-    Options options = new Options();
-    options.addOption("h", "help", false, "show usage");
-    options.addOption(
-        Option.builder()
-            .longOpt("project_id")
-            .desc("Project id")
-            .hasArg()
-            .argName("PROJECT-ID")
-            .required(true)
-            .build());
-    CommandLine cl = (new DefaultParser()).parse(options, args);
-    if (cl.hasOption("help")) {
-      HelpFormatter formater = new HelpFormatter();
-      formater.printHelp("{@smokeTest.name}", options);
-    }
-    executeNoCatch(cl.getOptionValue("project_id"));
+    executeNoCatch(getProjectId());
   @else
     executeNoCatch();
   @end
 @end
 
+@private getProjectIdForTest()
+  private static String getProjectId() {
```

Comment:
Please do it in same way as you did it here https://github.com/neozwu/google-cloud-java/blob/b9520a481c9c8fbac9b032eb77e0912eca21742a/google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/TopicAdminSmokeTest.java#L105